### ISSUE TITLE
New component: Riak Metric Receiver

### ISSUE THREAD

**Issue by armstrmi** [open-telemetry/opentelemetry-collector-contrib#8364]
## The purpose and use-cases of the new component
Riak metrics will be collected through the REST API using the [/stats](https://docs.riak.com/riak/kv/2.2.3/developing/api/http/status.1.html) endpoint.

This Riak receiver will collect metrics for [3.x](https://github.com/basho/riak/releases) 

## Example configuration for the component

The following configuration settings are required:

* `username`
* `password`

The following configuration settings are optional:

* `endpoint` (default: http://localhost:8098): The URL of the node to be monitored.
* `collection_interval` (default = 60s): This receiver collects metrics on an interval. Valid time units are ns, us (or µs), ms, s, m, h.
* `tls` (defaults defined [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/README.md)): TLS control. By default insecure settings are rejected and certificate verification is on.
```yaml
receivers:
  riak:
    endpoint: http://localhost:8098
    username: otelu
    password: $RIAK_PASSWORD
    collection_interval: 60s
```

## Telemetry data types supported
Metrics

**Comment by djaglowski** [comment]
I'll sponsor this.

**Comment by djaglowski** [comment]
Closed by #8548
